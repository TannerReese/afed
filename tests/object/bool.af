# Test boolean arithmetic and operators

# Boolean arithmetic operations
-true = ``,
!false = `true`,
true + true = ``,
true + false = ``,
false - true = ``,
true * false = ``,
true * true = ``,

# Boolean logical operations
true || false = ``,
false && true = ``,
false && false || true = ``,
false && false + true = ``,

# Boolean unimplemented operators
true % false = ``,
false // true = ``,
true ^ true = ``,

# Ternary
if true 3 4 = ``,
if false (if true 1 2) (if true 3 4) + 2 = ``,

# Ternary unimplemented operators
if + if = ``,
if * 3 = ``,

