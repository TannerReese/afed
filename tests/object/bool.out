# Test boolean arithmetic and operators

# Boolean arithmetic operations
-true = `false`,
!false = `true`,
true + true = `false`,
true + false = `true`,
false - true = `true`,
true * false = `false`,
true * true = `true`,

# Boolean logical operations
true || false = `true`,
false && true = `false`,
false && false || true = `true`,
false && false + true = `false`,

# Boolean unimplemented operators
true % false = `Eval Error: Binary operator % not implemented between true and false`,
false // true = `Eval Error: Binary operator // not implemented between false and true`,
true ^ true = `Eval Error: Binary operator ^ not implemented between true and true`,

# Ternary
if true 3 4 = `3`,
if false (if true 1 2) (if true 3 4) + 2 = `5`,

# Ternary unimplemented operators
if + if = `Eval Error: Binary operator + not implemented between if and if`,
if * 3 = `Eval Error: Binary operator * not implemented between if and 3`,

